frontend/src/v5/ui/routes/dashboard/projects/projectsList/createProjectModal/createProjectModal.component.tsx

```diff
@@ -0,0 +1,139 @@
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { useEffect, useState } from 'react';
+import { formatMessage } from '@/v5/services/intl';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { FormModal } from '@controls/modal/formModal/formDialog.component';
+import { useForm, SubmitHandler } from 'react-hook-form';
+import { ProjectsActionsDispatchers } from '@/v5/services/actionsDispatchers/projectsActions.dispatchers';
+import { CreateProjectSchema } from '@/v5/validation/projectSchemes/projectsSchemes';
+import { FormTextField } from '@controls/formTextField/formTextField.component';
+import { FormSelect } from '@controls/formSelect/formSelect.component';
+import { MenuItem } from '@mui/material';
+import { TeamspacesHooksSelectors } from '@/v5/services/selectorsHooks/teamspacesSelectors.hooks';
+import { isNetworkError, projectAlreadyExists } from '@/v5/validation/errors.helpers';
+import { NetworkError } from '@controls/errorMessage/networkError/networkError.component';
+import { UnexpectedError } from '@controls/errorMessage/unexpectedError/unexpectedError.component';
+
+interface ICreateProject {
+	open: boolean;
+	onClickClose: () => void;
+}
+
+interface IFormInput {
+	projectName: string;
+	teamspace: string;
+}
+
+export const CreateProjectForm = ({ open, onClickClose }: ICreateProject) => {
+	const teamspaces = TeamspacesHooksSelectors.selectTeamspaces();
+	const currentTeamspace = TeamspacesHooksSelectors.selectCurrentTeamspace();
+
+	const [existingProjectsByTeamspace, setExistingProjectsByTeamspace] = useState({});
+	const [isSubmitting, setIsSubmitting] = useState(false);
+	const [hasUnexpectedError, setHasUnexpectedError] = useState(false);
+
+	const DEFAULT_VALUES = {
+		teamspace: currentTeamspace,
+		projectName: '',
+	};
+
+	const {
+		control,
+		formState,
+		formState: { errors },
+		reset,
+		handleSubmit,
+		getValues,
+		trigger,
+	} = useForm<IFormInput>({
+		mode: 'onChange',
+		resolver: yupResolver(CreateProjectSchema(existingProjectsByTeamspace)),
+		defaultValues: DEFAULT_VALUES,
+	});
+
+	const onSubmissionError = (error) => {
+		setHasUnexpectedError(false);
+		if (isNetworkError(error)) return;
+		if (projectAlreadyExists(error)) {
+			const { projectName, teamspace } = getValues();
+			setExistingProjectsByTeamspace({
+				...existingProjectsByTeamspace,
+				[teamspace]: [
+					...(existingProjectsByTeamspace[teamspace] || []),
+					projectName,
+				],
+			});
+			return;
+		}
+		setHasUnexpectedError(true);
```

Comment:
An "unexpected error" that I managed to get which we should probably cover is a 401 Unauthorised. If you add a user to a teamspace but don't give them admin then that user can't make projects in that teamspace

Reply:
Admin related issue will be addressed in another branch